### PR TITLE
pex: allow disabled pex reactor

### PR DESCRIPTION
**Pull request by tychoish**
This ensures the implementation respects disabling the pex reactor.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -576,8 +582,10 @@ func (n *nodeImpl) OnStart() error {
 		}
 	}
 
-	if err := n.pexReactor.Start(); err != nil {
-		return err
+	if n.config.P2P.PexReactor {
+		if err := n.pexReactor.Start(); err != nil {
+			return err
+		}
```
Comment (cmwaters) [inline node/node.go:588]: Is it necessary to check this here?

**Review by cmwaters — APPROVED** [review]
LGTM

File: node/node.go (hunk 1 above)
Comment (tychoish) [reply node/node.go:588]: certainly safer. regardless we need to check if it's nil before we start it in any case, and given that `n.pexReactor` is an interface type, the possibility of the field being non-nil while holding a nil implementation exists, and checking the config value is a safer option.

Comment (cmwaters) [reply node/node.go:588]: can the pex reactor be `nil`?

Comment (cmwaters) [reply node/node.go:588]: Ah sorry I got confused. I thought this method was just for the seed node but then I realised that both the full node and seed node share the same implementation. Ignore this comment

**Pull request merged**